lib/le/host/connection.rb

```diff
@@ -1,146 +1,138 @@
-#!      attr_accessor :token, :queue, :started, :thread, :conn, :local, :debug, :ssl, :datahub_enabled, :dathub_ip, :datahub_port, :host_id, :custom_host, :host_name_enabled, :host_name
-      attr_accessor :token, :region,:queue, :started, :thread, :conn, :local, :debug, :ssl, :datahub_enabled, :datahub_ip, :datahub_port, :datahub_endpoint, :host_id, :host_name_enabled, :host_name, :custom_host, :udp_port, :use_data_endpoint
-
-
-      def initialize(token, region, local, debug, ssl, datahub_endpoint, host_id, custom_host, udp_port, use_data_endpoint)
-          if local
-            device = if local.class <= TrueClass
-              if defined?(Rails)
-                Rails.root.join("log","#{Rails.env}.log")
-              else
-                STDOUT
-              end
-            else
-            local
-            end
+      attr_accessor :token, :region, :queue, :started, :thread, :conn, :local,
+                    :debug, :ssl, :datahub_enabled, :datahub_ip, :datahub_port,
+                    :datahub_endpoint, :host_id, :host_name_enabled, :host_name,
+                    :custom_host, :udp_port, :use_data_endpoint
+
+      def initialize(token, region, local, debug, ssl, datahub_endpoint,
+                     host_id, custom_host, udp_port, use_data_endpoint)
+        if local
+          device = if local.class <= TrueClass
+                     if defined?(Rails)
+                       Rails.root.join('log', "#{Rails.env}.log")
+                     else
+                       STDOUT
+                     end
+                   else
+                     local
+                   end
           @logger_console = Logger.new(device)
-          end
+        end
 
-          @region = region
-          @local = !!local
-          @debug= debug
-          @ssl = ssl
-          @udp_port = udp_port
-          @use_data_endpoint = use_data_endpoint
+        @region = region
+        @local = local.nil? || local == false ? false : true # Replace "!!"
+        @debug = debug
+        @ssl = ssl
+        @udp_port = udp_port
+        @use_data_endpoint = use_data_endpoint
 
         @datahub_endpoint = datahub_endpoint
         if !@datahub_endpoint[0].empty?
-          @datahub_enabled=true
-          @datahub_ip="#{@datahub_endpoint[0]}"
-          @datahub_port=@datahub_endpoint[1]
+          @datahub_enabled = true
+          @datahub_ip = (@datahub_endpoint[0]).to_s
+          @datahub_port = @datahub_endpoint[1]
         else
-          @datahub_enabled=false
+          @datahub_enabled = false
         end
 
-
-        if (@datahub_enabled && @ssl)
-          puts ("\n\nYou Cannot have DataHub and SSL enabled at the same time.  Please set SSL value to false in your environment.rb file or used Token-Based logging by leaving the Datahub IP address blank. Exiting application. \n\n")
+        if @datahub_enabled && @ssl
+          puts("\n\nYou Cannot have DataHub and SSL enabled at the same time.
```

Comment:
Yikes. It sure is lucky we didn't get acquired by a security company... 😅 We should probably think about fixing that...